app/models/user.rb

```diff
@@ -1,10 +1,12 @@
 class User < ApplicationRecord
+  has_many :results, dependent: :destroy
+  has_many :created_tests, class_name: 'Test', foreign_key: 'author_id', dependent: :nullify
+  has_many :processed_tests, class_name: 'Test', through: :results, source: 'user'
```

Comment:
```suggestion
  has_many :results, dependent: :destroy
  has_many :processed_tests, class_name: 'Test', through: :results, source: 'user'
  has_many :created_tests, class_name: 'Test', foreign_key: 'author_id', dependent: :nullify
```
в прошлый раз оставлял комментарий по этому поводу